clap_derive/src/utils/doc_comments.rs

```diff
@@ -117,10 +83,200 @@ fn is_blank(s: &str) -> bool {
+        styles: Vec<Style>,
+    }
+
+    impl MarkdownWriter {
+        fn newline(&mut self) {
+            self.output.push('\n');
+        }
+        fn endline(&mut self) {
+            if !self.output.ends_with('\n') {
+                self.newline();
+            }
+        }
+
+        fn write_fmt(&mut self, arguments: fmt::Arguments<'_>) {
+            if self.output.ends_with('\n') {
+                write!(self.output, "{0: <1$}", "", self.indentation).unwrap();
+            }
+            self.output.write_fmt(arguments).unwrap();
+        }
+
+        fn start_link(&mut self, dest_url: pulldown_cmark::CowStr<'_>) {
+            write!(self, "\x1B]8;;{dest_url}\x1B\\");
+        }
+        fn end_link(&mut self) {
+            write!(self, "\x1B]8;;\x1B\\");
+        }
+
+        fn start_style(&mut self, style: Style) {
+            self.styles.push(style);
+            write!(self, "{style}");
+        }
+        fn end_style(&mut self, style: Style) {
+            let last_style = self.styles.pop();
+            debug_assert_eq!(last_style.unwrap(), style);
+
+            write!(self, "{Reset}");
+            // Reapplying all, because anstyle doesn't support merging styles
+            // (probably because the ambiguity around colors)
+            // TODO If we decide not to support any colors, we can replace this with
+            // anstyle::Effects and remove the need for applying them all individually.
+            for style in &self.styles {
+                write!(self.output, "{style}").unwrap();
+            }
+        }
+    }
+
+    pub(super) fn parse_markdown(input: &[String]) -> (String, Option<String>) {
+        // Markdown Configuration
+        let parsing_options = Options::ENABLE_STRIKETHROUGH /* TODO UNICODE | Options::ENABLE_SMART_PUNCTUATION */;
+        // Minimal Styling for now, because we cannot configure it
+        let style_heading = Style::new().bold();
+        let style_emphasis = Style::new().italic();
+        let style_strong = Style::new().bold();
+        let style_strike_through = Style::new().strikethrough();
+        let style_link = Style::new().underline();
+        // TODO decide how to style code
+        let style_code = Style::new().dimmed();
+        let tab_width = 2;
+        // TODO UNICODE let list_symbol = '•';
+        let list_symbol = '-';
+
+        let input = input.join("\n");
+        let input = Parser::new_ext(&input, parsing_options);
+
+        let mut short = None;
+
+        let mut writer = MarkdownWriter::default();
+
+        let mut list_indices = Vec::new();
+
+        for event in input {
+            match event {
+                Event::Start(Tag::Paragraph) => { /* nothing to do */ }
+                Event::Start(Tag::Heading { .. }) => writer.start_style(style_heading),
+                Event::Start(
+                    Tag::Image { .. } | Tag::BlockQuote(_) | Tag::CodeBlock(_) | Tag::HtmlBlock,
+                ) => { /* IGNORED */ }
```

Comment:
We should do something for block quotes and code blocks

For block quotes, we could
- Indent
- Indent with a leading pipe (to look like how its commonly rendered)
- Indent with a leading `>`

For code blocks, we could
- Indent and apply the `style_code`
- Add on a leading pipe or other decor